php/libraries/LorisForm.class.inc

```diff
@@ -677,14 +677,26 @@ class LorisForm
     protected function selectHTML($el)
     {
         $strOptions    = '';
-        $options       = $el['options'];
+        $options       = $el['options'] ?? array();
         $cls           = '';
         $changehandler = '';
         $multiple      = '';
         $id            = '';
         $disabled      = '';
-        $name          = $el['name'];
-
+        $name          = $el['name'] ?? '';
+
+        if (!is_array($options) || !($options instanceof Traversable)) {
```

Comment:
I don't think this logic is right. 'not' isn't a commutative property in boolean logic. This should either be '&&', or the 'not' needs to cover the whole '||' rather than the individual pieces.

(I'm also not sure if you even need the is_array part. Does array implement Traversable?)